src/network/networkd-route.c

```diff
@@ -299,9 +299,10 @@ void route_hash_func(const Route *route, struct siphash *state) {
                 siphash24_compress(&route->src, FAMILY_ADDRESS_SIZE(route->family), state);
 
                 siphash24_compress(&route->gw_family, sizeof(route->gw_family), state);
-                if (IN_SET(route->gw_family, AF_INET, AF_INET6))
+                if (IN_SET(route->gw_family, AF_INET, AF_INET6)) {
                         siphash24_compress(&route->gw, FAMILY_ADDRESS_SIZE(route->gw_family), state);
-
+                        siphash24_compress(&route->gw_weight, sizeof(route->gw_weight), state);
+                }
```

Comment:
Hmm, why {} ?

Reply:
because there are now two expressios in the {}